### PR TITLE
Telegram Bot: Call the tweepy auth function to the telegram class

### PR DESCRIPTION
**Pull request by iammarco11**
Please check the code and suggest me how to do it. The code is incorrect

### CODE REVIEW

**Review by Aniketh01 — COMMENTED** [review]
Is this changes tested? Is it working for you?

File: tg_bot.py [hunk 1]
```diff
@@ -20,19 +21,23 @@ class Telegram_Bot(object):
     def __init__(self, dirpath):
         self.dirpath = dirpath
 
+    def tweepy_auth(self):
+        api = Twitter_Bot.nawab_twitter_authenticate()
```
Comment (Aniketh01) [inline tg_bot.py]: 'nawab_twitter_authenticate()' does the tweepy authentication for you, Why wrapping it up again in a function?

File: tg_bot.py [hunk 2]
```diff
@@ -20,19 +21,23 @@ class Telegram_Bot(object):
...
         updater = Updater(token=tg.token, use_context=True)
         return updater
 
     def start(self, update, context):
-        auth = tweepy.OAuthHandler(config.consumer_key, config.consumer_secret)
+        """auth = tweepy.OAuthHandler(config.consumer_key, config.consumer_secret)
         auth.set_access_token(config.access_token_key,
                               config.access_token_secret)
-        api = tweepy.API(auth)
+        api = tweepy.API(auth)"""
```
Comment (Aniketh01) [inline tg_bot.py]: Rather than commenting, remove it off!

File: tg_bot.py [hunk 3]
```diff
@@ -49,16 +54,16 @@ def start(self, update, context):
...
         # CallbackQueries need to be answered, even if no notification to the user is needed
         # Some clients may have trouble otherwise. See https://core.telegram.org/bots/api#callbackquery
         query.answer()
         data = query.data
         try:
-            api.retweet(data)
+            self.api.retweet(data)
```
Comment (Aniketh01) [inline tg_bot.py]: Does this work for you?? I doubt it.

**Comment by iammarco11** [comment]
It was tested but was not working. Error was api was undefined.

**Review by Aniketh01 — CHANGES_REQUESTED** [review]
> It was tested but was not working. Error was api was undefined.

Fix it, please. You need to understand how classes work and how you can import that particular function can be imported to tg bot class. Find similarities in nawab.py. Perhaps you might have to re-write the twitter class definition itself. Keep an open mind :)

Try to read a bit on class and objects in python and how to call those methods and stuff. General OOP concepts would do well too :)

File: tg_bot.py (hunk 3 above)
Comment (iammarco11) [reply tg_bot.py]: it works

File: tg_bot.py [hunk 4]
```diff
@@ -17,22 +17,19 @@
 
 class Telegram_Bot(object):
 
-    def __init__(self, dirpath):
+    def __init__(self, api, dirpath):
         self.dirpath = dirpath
+        self.api = api
```
Comment (Aniketh01) [inline tg_bot.py]: Change the variable name to Twitter api or something. Can be ambiguous later

File: nawab.py [hunk 5]
```diff
@@ -16,8 +16,10 @@ def twitter_bot_run(data, dirpath):
     tw_bot.nawab_retweet_tweet(api)
 
 
-def tg_bot_run(dirpath):
-    bot = tg_bot.Telegram_Bot(dirpath)
+def tg_bot_run(data, dirpath):
+    tbot = twitter_bot.Twitter_Bot(data, dirpath)
```
Comment (Aniketh01) [inline nawab.py]: Follow the similar naming convention as the twitter run function

Comment (Aniketh01) [reply nawab.py]: The Twitter class objects are created twice in this file alone. See if there is a way to clean that up and have just a single instance.

**Comment by Aniketh01** [comment]
In a overall, look good. I will test it out later today and will merge accordingly.

**Review by Aniketh01 — CHANGES_REQUESTED** [review]
There are few inline reviews still pending.

**Review by Aniketh01 — COMMENTED** [review]
I think the buttons all fail now?

I also found caused error "'tuple' object has no attribute 'get_status'"

**Pull request merged**